discord/client.py

```diff
@@ -1498,13 +1505,24 @@ def edit_profile(self, password, **fields):
             else:
                 avatar = None
 
-        payload = {
-            'password': password,
-            'new_password': fields.get('new_password'),
-            'email': fields.get('email', self.email),
-            'username': fields.get('username', self.user.name),
-            'avatar': avatar
-        }
+        if self.is_bot_account:
+            payload = {
+                'username': fields.get('username', self.user.name),
+                'avatar': avatar
+            }
+        else:
+            try:
+                password = fields['password']
+            except KeyError:
+                raise("Missing password argument.")
```

Comment:
This will actually raise a TypeError. You should either let the error pass through or re-raise it with the custom text.